From e68810306840d90c64bb3c9b6db6630d748de262 Mon Sep 17 00:00:00 2001
From: Michael Brown <mbrown@fensystems.co.uk>
Date: Sun, 22 Sep 2019 21:51:29 +0100
Subject: [PATCH 545/587] bcm2708_fb: Fix layout of struct
 vc4_display_settings_t

The display parameters returned by the VC4 firmware in response to the
RPI_FIRMWARE_FRAMEBUFFER_GET_DISPLAY_SETTINGS tag do not match the
layout of struct vc4_display_settings_t: the colour depth and row
pitch are erroneously swapped in the kernel definition.  This can
trigger a kernel warning from pixel_to_pat(), such as:

  pixel_to_pat(): unsupported pixelformat 7296

Fix by adjusting the layout of struct vc4_display_settings_t to match
the layout as used by the VC4 firmware.

Signed-off-by: Michael Brown <mbrown@fensystems.co.uk>
---
 drivers/video/fbdev/bcm2708_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/bcm2708_fb.c b/drivers/video/fbdev/bcm2708_fb.c
index 4da577c8a0d8..59443f79f228 100644
--- a/drivers/video/fbdev/bcm2708_fb.c
+++ b/drivers/video/fbdev/bcm2708_fb.c
@@ -88,8 +88,8 @@ struct vc4_display_settings_t {
 	u32 display_num;
 	u32 width;
 	u32 height;
-	u32 pitch;
 	u32 depth;
+	u32 pitch;
 	u32 virtual_width;
 	u32 virtual_height;
 	u32 virtual_width_offset;
-- 
2.17.1

